### PR TITLE
Added bank plugin (intial version)

### PR DESCRIPTION
**Pull request by TheLonelyDev**
Added bank plugin (intial version)
Added check to getItemPrice in ItemManager so 404 requests do not "kill" the future requests (die).

Issue, https://github.com/runelite/runelite/issues/349

We might need to look into a way to send bulk requests to the runelite.net API to make calculations faster on the client and so it is a bit friendlier in terms of amount of requests.

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/bank/BankPlugin.java [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+@PluginDescriptor(name = "Bank Value")
+@Slf4j
+public class BankPlugin extends Plugin
+{
+	private static final float HIGH_ALCHEMY_CONSTANT = 0.6f;
+	private static final int ITEM_EMPTY = 6512;
+	private static final int ITEM_COINS = 617;
+
+	@Inject
+	Client client;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/bank/BankPlugin.java]: All the injected fields should be private and have new lines above annotations

File: runelite-client/src/main/java/net/runelite/client/plugins/bank/BankPlugin.java [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+	@Inject
+	BankConfig config;
+	@Inject
+	ItemManager itemManager;
+	@Inject
+	ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor();
+
+	private String strCurrentTab;
+	private String[] suffixes = {"", "K", "M", "B", "T", "Q", "W", "T", "F"};
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/bank/BankPlugin.java]: This should be constant

File: runelite-client/src/main/java/net/runelite/client/plugins/bank/BankPlugin.java [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+
+	private ScheduledFuture futureCalculateBank;
+	private Future taskWidget;
+	private BankRunnable runCalculateBank = new BankRunnable();
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/bank/BankPlugin.java]: This field should be final

File: runelite-client/src/main/java/net/runelite/client/plugins/bank/BankPlugin.java [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+	{
+		return configManager.getConfig(BankConfig.class);
+	}
+
+	@Subscribe
+	public void checkBankState(ConfigChanged event)
+	{
+		if (!config.isEnabled() || client.getWidget(WidgetInfo.BANK_ITEM_COUNT) == null) return;
+
+		CancelTasks();
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/bank/BankPlugin.java]: follow Java code conventions for naming methods (camelCase)

File: runelite-client/src/main/java/net/runelite/client/plugins/bank/BankPlugin.java [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+
+			double dblTick = System.currentTimeMillis() + 10000;
+
+			taskWidget = executor.scheduleAtFixedRate(() -> WaitForWidget(WidgetInfo.BANK_ITEM_COUNT, dblTick), 0, 1, TimeUnit.SECONDS);
+		}
+
+		return;
+	}
+
+	private void WaitForWidget(WidgetInfo info, double dblTick)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/bank/BankPlugin.java]: Follow java coding conventions for naming methods

File: runelite-client/src/main/java/net/runelite/client/plugins/bank/BankPlugin.java [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+			log.warn("Error looking up item price", e);
+			return new int[]{0, 0};
+		}
+
+		int itemCompositionPrice = itemComposition.getPrice();
+
+		return new int[]{itemPrice == null ? 0 : itemPrice.getPrice() * quantity, itemCompositionPrice <= 0 ? 0 : Math.round(itemCompositionPrice * HIGH_ALCHEMY_CONSTANT) * quantity};
+	}
+
+	private java.util.List<Widget> GetItems()
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/bank/BankPlugin.java]: Follow java naming convetions for naming methods

File: runelite-client/src/main/java/net/runelite/client/plugins/bank/BankPlugin.java [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+			if (widgetChildren[i].getItemId() == ITEM_EMPTY || widgetChildren[i].getItemId() <= 0 || widgetChildren[i].isHidden())
+				continue;
+
+			widgetItems.add(widgetChildren[i]);
+		}
+
+		return widgetItems;
+	}
+
+	private String formatNumber(double n)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/bank/BankPlugin.java]: Both formatNumber methods should be static

Comment (deathbeam) [reply runelite-client/src/main/java/net/runelite/client/plugins/bank/BankPlugin.java]: Also, use the ItemManager#quantityToStackSize instead (and modify that method to include also bills etc.

File: runelite-client/src/main/java/net/runelite/client/plugins/bank/BankPlugin.java [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+
+		public void run()
+		{
+			try
+			{
+				boolExit = false;
+
+				Widget widgetBankTitleBar = client.getWidget(WidgetInfo.BANK_TITLE_BAR);
+
+				if (widgetBankTitleBar == null || widgetBankTitleBar.isHidden()) return;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/bank/BankPlugin.java]: Dont use conditions without braces

File: runelite-client/src/main/java/net/runelite/client/plugins/bank/BankPlugin.java [hunk 9]
```diff
@@ -0,0 +1,234 @@
...
+
+					strCurrentTab = config.customName() + " ~" + formatNumber(intGEValue) + " (GE)";
+
+					if (config.showHA()) strCurrentTab += " ~" + formatNumber(intHAValue) + " (HA)";
+
+					widgetBankTitleBar.setText(strCurrentTab);
+				}
+			} catch (Exception e)
+			{
+
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/bank/BankPlugin.java]: Never use empty catch statements

File: runelite-client/src/main/java/net/runelite/client/plugins/bank/BankPlugin.java [hunk 10]
```diff
@@ -0,0 +1,234 @@
...
+			}
+		}
+
+		public void abort()
+		{
+			boolExit = true;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/bank/BankPlugin.java]: Just cancel futures, remove the abortion method from runnable

**Comment by Adam-** [comment]
Done in #824

**Pull request closed without merging**